app/src/main/java/com/x8bit/bitwarden/data/platform/manager/ResourceCacheManagerImpl.kt

```diff
@@ -0,0 +1,36 @@
+package com.x8bit.bitwarden.data.platform.manager
+
+import android.content.Context
+import com.x8bit.bitwarden.R
+
+/**
+ * Primary implementation of [ResourceCacheManager].
+ */
+class ResourceCacheManagerImpl(
+    val context: Context,
+) : ResourceCacheManager {
+    private var cachedDomainExceptionSuffixes: List<String>? = null
+    private var cachedDomainNormalSuffixes: List<String>? = null
+    private var cachedDomainWildCardSuffixes: List<String>? = null
+
+    override val domainExceptionSuffixes: List<String>
+        get() = cachedDomainExceptionSuffixes ?: context
+            .resources
+            .getStringArray(R.array.exception_suffixes)
+            .toList()
+            .apply { cachedDomainExceptionSuffixes = this }
```

Comment:
Can we do this instead:

```kotlin
 .also { cachedDomainExceptionSuffixes = it }
```

We typically want to reserve `apply` for cases where the `this` can be implied.